lib/wallet/http.js

```diff
@@ -1427,6 +1427,99 @@ class HTTP extends Server {
         errorMessages: errorMessages});
     });
 
+    // Create Batch Finish (Redeem or Register)
+    this.post('/wallet/:id/batch/finish', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      let names = valid.array('names');
+      const passphrase = valid.str('passphrase');
+
+      assert(names && names.length > 0, 'Names are required.');
+
+      names.forEach((element) => {
+        const validator = new Validator(element);
+        if (!validator.has('name') || !validator.has('data'))
+           throw new Error('name and data must be present in every element.');
+
+        validator.str('name');
+        validator.obj('data');
+
+        const dataValidator = new Validator(element.data);
+        if (!dataValidator.has('records'))
+           throw new Error('data must be in form: {records: []}');
+
+        validator.array(element.data.records);
+      });
+
+      names = names.map(({name, data}) => {
+        return {name, data: Resource.fromJSON(data)};
+      });
+
+      const options = TransactionOptions.fromValidator(valid);
+
+      const finishCache = req.wallet.sendFinishResults;
+
+      const processedFinishes = [];
+      let errorMessages = [];
+      const uniqueNames = [];
+
+      names.forEach(({name, data}) => {
+        const cachedNames = finishCache.get(name);
+        if (!cachedNames) {
+          uniqueNames.push({name, data});
+        } else {
+          cachedNames.forEach(cachedName => processedFinishes.push(cachedName));
+        }
+      });
+
+      if (uniqueNames.length > 0) {
+        const nameHashMap = new Map();
+        uniqueNames.forEach(({name, data}) => {
+          const nameHash = rules.hashName(name).toString('hex');
+          nameHashMap.set(nameHash, name);
+        });
+
+        // always broadcast
+        const {mtx, errorMessages: errors} = await req.wallet
+           .createBatchFinish(uniqueNames, options);
+
+        errorMessages = errors;
```

Comment:
What is this line of code for? Can we just leave line 1482 as `const { mtx, errorMessages }` if we want the variable to be named that